### PR TITLE
Moved example into owner package. 

### PR DESCRIPTION
**Pull request by asciiu**
Renamed example package to owner and implemented the owner accounts api method.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -13,3 +13,7 @@
 
 # General
 .DS_Store
+
+# Local dev
+/config/test.env
+owner_test.go
```
Comment (quentinlesceller) [inline .gitignore:19]: Why add this? Seems like a local conf?

File: README.MD [hunk 2]
```diff
@@ -9,7 +9,7 @@ Currently, it contains the basic consensus parameters, chain, slate structures a
 
 ## Requirements
 
-[Go](http://golang.org) 1.11 or newer.
+[Go](http://golang.org) 1.14 or newer.
```
Comment (quentinlesceller) [inline README.MD:12]: 👍

File: owner/secure_owner_api.go [hunk 3]
```diff
@@ -140,6 +140,48 @@ func (owner *SecureOwnerAPI) InitSecureAPI(pubKey []byte) (string, error) {
...
+		return nil, err
+	}
+
+	envl, err := owner.client.EncryptedRequest("accounts", paramsBytes, owner.sharedSecret)
+	if err != nil {
+		return nil, err
+	}
+
+	if envl == nil {
+		return nil, errors.New("Accounts: Empty RPC Response from grin-wallet")
```
Comment (quentinlesceller) [inline owner/secure_owner_api.go:162]: Being a bit picky but can you change it to the following so we keep it consistent?
```suggestion
		return nil, errors.New("OwnerAPI: Empty RPC Response from grin-wallet")
```

File: owner/secure_owner_api.go [hunk 4]
```diff
@@ -140,6 +140,48 @@ func (owner *SecureOwnerAPI) InitSecureAPI(pubKey []byte) (string, error) {
 	return serverPubKey, nil
 }
 
+// Accounts Returns a list of accounts stored in the wallet
+// (i.e. mappings between user-specified labels and BIP32 derivation paths
```
Comment (quentinlesceller) [inline owner/secure_owner_api.go:144]: very nice 👍

File: libwallet/api_impl_types.go [hunk 5]
```diff
@@ -20,6 +20,11 @@ import (
 	"github.com/blockcypher/libgrin/libwallet/slateversions"
 )
 
+type AccountPathMapping struct {
+	Label string
+	Path  string
+}
+
```
Comment (quentinlesceller) [inline libwallet/api_impl_types.go:27]: If we follow Grin structure this should be in `libwallet/types.go` https://github.com/mimblewimble/grin-wallet/blob/master/libwallet/src/types.rs#L907 also can you add the comment about what it is?

File: owner/secure_owner_api_test.go [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+	"testing"
+
+	log "github.com/sirupsen/logrus"
+)
+
+func checkError(err error) {
+	if err != nil {
+		log.Fatal(err)
+	}
+}
```
Comment (quentinlesceller) [inline owner/secure_owner_api_test.go:27]: Think you forgot to delete some test functions :)
```suggestion
)
```

Comment (quentinlesceller) [reply owner/secure_owner_api_test.go:27]: I think you can just checkout this file for now. Would be great to have some real test with a CI.
Like download grin and grin-wallet binaries and test the functions.

File: .gitignore (hunk 1 above)
Comment (asciiu) [reply .gitignore:19]: It's in the ignore so my local env file does not ever get checked in. The last thing I want is to check in any sensitive env params that I'm using my development by mistake. Plus using env config is useful for test.env, stage.env, or production.env.

**Comment by asciiu** [comment]
I figured moving this stuff into the "owner" package and naming the other client lib "foreign" to match the rust api. I'm fine with whatever naming convention you want though since this isn't my project.

**Pull request merged**